modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -124,6 +126,45 @@ public Product getChannelProduct(
 		return _toProduct(commerceContext, cpDefinition);
 	}
 
+	@Override
+	public Product getChannelProductByFriendlyUrlPath(
+			Long channelId, String friendlyUrl, Long accountId)
+		throws Exception {
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannel(channelId);
+
+		Long commerceAccountId = _getCommerceAccountId(
+			accountId, commerceChannel);
+
+		if (!_isAccountEntryEligible(
+				commerceAccountId, commerceChannel.getCommerceChannelId())) {
+
+			return null;
+		}
+
+		Group companyGroup = _groupLocalService.getCompanyGroup(
+			commerceChannel.getCompanyId());
```

Comment:
Why do you retrieve the Group when you can use `commerceChannel.getGroupId() `?

Reply:
are different: 
companyGroup.getGroupId() =  20120
commerceChannel.getGroupId() = 36017

Reply:
> are different: 
> companyGroup.getGroupId() = 20120
> commerceChannel.getGroupId() = 36017

And if you try to use `commerceChannel.getSiteGroupId()`

Reply:
I tried
```
commerceChannel.getSiteGroupId(): 33815
companyGroup.getGroupId(): 20120
```